src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -374,7 +388,34 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                     var analyzerOptions = new AnalyzerOptions(ImmutableArray<AdditionalText>.CastUp(additionalTextFiles));
 
                     analyzerDriver = AnalyzerDriver.CreateAndAttachToCompilation(compilation, analyzers, analyzerOptions, analyzerManager, addExceptionDiagnostic, Arguments.ReportAnalyzer, out compilation, analyzerCts.Token);
-                    getAnalyzerDiagnostics = () => analyzerDriver.GetDiagnosticsAsync(compilation).Result;
+                    reportAnalyzer = Arguments.ReportAnalyzer && !analyzers.IsEmpty;
+                }
+
+                Compilation newCompilation = null;
+                if ((analyzerDriver != null) || includeGenerators)
+                {
+                    getAnalyzerDiagnostics = (diags, result) =>
+                    {
+                        if (result && (analyzerDriver != null))
+                        {
+                            var hostDiagnostics = analyzerDriver.GetDiagnosticsAsync(compilation).Result;
```

Comment:
Not sure why you are running diagnostics before adding generated source.. Won't this always/often cause errors to existing in the users source?